### PR TITLE
feat: implemented logic to output diagnostics log by adding additional output flag to validate cli command.

### LINKED ISSUE #1095: Validate command output is polluted by noisy messages
#### Describe the bug

Validate command supports multiple diagnostics formats. 
I assume that formats like json or junit are intended to be processed by machines.

However, validate command standard output is a mix of the selected diagnostics format and informational messages. 
User must parse / filter standard output to end up with the expected file.

#### How to Reproduce

```
docker run -it --rm     -v ${PWD}:/spec test asyncapi validate /spec/docs/events/asyncapi.yaml --diagnostics-format=junit  > asyncapi.xml

cat asyncapi.xml
(node:1) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)

File /spec/docs/events/asyncapi.yaml and/or referenced documents have governance issues.
<?xml version="1.0" encoding="utf-8"?>
<testsuites>
<testsuite package="org.spectral" time="0" tests="1" errors="0" failures="1" name="/spec/docs/events/asyncapi.yaml">
<testcase time="0" name="org.spectral.asyncapi-document-resolved(#/channels/hems-site-events/messages/GatewayConnectionMqttDisconnected/payloa)" classname="/spec/docs/events/asyncapi"><failure message="Property &quot;payloa&quot; is not expected to be here"><![CDATA[line 94, col 16, Property "payloa" is not expected to be here (asyncapi-document-resolved) at path #/channels/site-events/messages/GatewayConnectionMqttDisconnected/payloa]]></failure></testcase>
</testsuite>
</testsuites>
```

#### Expected behavior

It would be nice to separate command logs from actual output. 
Those options could work:
 * Add an optional outputFile parameter 
 * Add an optional silent parameter

### PR DESCRIPTION
**Pull request by SahilDahekar**
**Description**

- Added additional output flag param ```-o``` to output diagnostics log to a file.
- By default the diagnostics log will be printed in terminal.

**Related issue(s)**
Resolves #1095

### CODE REVIEW

**Comment by Souvikns** [comment]
@SahilDahekar you have some lint issues. Can you fix them, so that we can start with the review?

**Comment by SahilDahekar** [comment]
Hey @Souvikns , thanks for the response.  I have fixed the lint issues now and we can go ahead with review.

**Review by Souvikns — CHANGES_REQUESTED** [review]
Everything looks good just remove the commented code and we are good to go.

File: src/core/parser.ts [hunk 1]
```diff
@@ -76,30 +99,92 @@ export async function parse(command: Command, specFile: Specification, options:
...
+//       } else {
+//         command.log(diagnosticsOutput);
+//       }
+//     }
+//   } else if (logDiagnostics) {
+//     command.log(`\n${sourceString} is valid! ${sourceString} and referenced documents don't have governance issues.`);
+//   }
+
+//   return ValidationStatus.VALID;
+// }
```
[the remark this answers is not in the document]
Comment (SahilDahekar) [reply src/core/parser.ts:136]: Yes

**Review by Souvikns — APPROVED** [review]

**Pull request merged**